lte/gateway/c/oai/tasks/nas5g/src/AmfMessage.cpp

```diff
@@ -0,0 +1,288 @@
+/*
+   Copyright 2020 The Magma Authors.
+   This source code is licensed under the BSD-style license found in the
+   LICENSE file in the root directory of this source tree.
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+#include "AmfMessage.h"
+#include "M5gNasMessage.h"
+#include "M5GCommonDefs.h"
+
+using namespace std;
+namespace magma5g {
+AmfMsg::AmfMsg(){};
+AmfMsg::~AmfMsg(){};
+
+// Decode AMF NAS Header and Message
+int AmfMsg::M5gNasMessageDecodeMsg(AmfMsg* msg, uint8_t* buffer, uint32_t len) {
+  int header_result = 0;
+  int decode_result = 0;
+
+  if (len > 0 || buffer != NULL) {
+    header_result = msg->AmfMsgDecodeHeaderMsg(&msg->header, buffer, len);
+    if (header_result <= 0) {
+      MLOG(MERROR) << "   Error : Header Decoding Failed" << std::dec
+                   << RETURN_ERROR;
+      return (RETURN_ERROR);
+    }
+  } else {
+    MLOG(MERROR) << "Error : Buffer is Empty";
+    return (RETURN_ERROR);
+  }
+  MLOG(MDEBUG) << "   epd = 0x" << hex
+               << int(msg->header.extended_protocol_discriminator) << "\n"
+               << "   security hdr =  0x" << hex
+               << int(msg->header.sec_header_type) << "\n"
+               << "   hdr type = 0x" << hex << int(msg->header.message_type)
+               << "\n";
```

Comment:
switch to OAI_LOG so that we can see them in mme.log.

Reply:
we will change MLOGs to OAI_LOGs and raise a separate PR